File: ontrack-extension-git/src/main/java/net/nemerosa/ontrack/extension/git/client/GitDiffEntry.java
package net.nemerosa.ontrack.extension.git.client;

import lombok.Data;
import org.eclipse.jgit.revwalk.RevCommit;

@Data
public class GitDiffEntry {

    private final GitChangeType changeType;
    private final String oldPath;
    private final String newPath;

    public String getReferencePath() {
        switch (changeType) {
            case DELETE:
                return oldPath;
            case MODIFY:
            case RENAME:
            case COPY:
            case ADD:
            default:
                return newPath;
        }
    }

    public String getReferenceId(RevCommit from, RevCommit to) {
        switch (changeType) {
            case DELETE:
                return from.getId().name();
            case MODIFY:
            case RENAME:
            case COPY:
            case ADD:
            default:
                return to.getId().name();
        }
    }
}

File: ontrack-git/src/main/java/net/nemerosa/ontrack/git/model/GitDiffEntry.kt
package net.nemerosa.ontrack.git.model

import org.eclipse.jgit.revwalk.RevCommit

class GitDiffEntry(
        val changeType: GitChangeType,
        val oldPath: String,
        val newPath: String
) {

    val referencePath: String
        get() = when (changeType) {
            GitChangeType.DELETE -> oldPath
            else -> newPath
        }

    fun getReferenceId(from: RevCommit, to: RevCommit): String = when (changeType) {
        GitChangeType.DELETE -> from.id.name()
        else -> to.id.name()
    }
}
